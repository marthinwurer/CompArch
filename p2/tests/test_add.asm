;
; tests ADD
;
; uses instructions BPT, HALT
;
; uses addressing modes 0, 6
;
	.globl	start
;
one:	.word	1
n_one:	.word	-1
five:	.word	5
n_five:	.word	-5
;
start:	add	one,r1
	bpt
;
	add	five,r1
	bpt
;
	add	n_one,r1
	bpt
;
	add	n_five,r1
	bpt
;
	add	n_five,r2
	add	n_five,r2
	bpt
;
	add	five,r3
	add	five,r3
	bpt
;
	add	five,r4
	add	n_five,r4
	add	n_five,r4
	bpt
;
	add	n_five,r5
	add	five,r5
	add	five,r5
	bpt
;
	halt
;
	.end	start
