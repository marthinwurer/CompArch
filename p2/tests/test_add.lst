      1                                 ;
      2                                 ; tests ADD
      3                                 ;
      4                                 ; uses instructions BPT, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000001                  one:     .word   1
     11 000002  177777                  n_one:   .word   -1
     12 000004  000005                  five:    .word   5
     13 000006  177773                  n_five:  .word   -5
     14                                 ;
     15 000010  066701  177764          start:   add     one,r1
     16 000014  000003                           bpt
     17                                 ;
     18 000016  066701  177762                   add     five,r1
     19 000022  000003                           bpt
     20                                 ;
     21 000024  066701  177752                   add     n_one,r1
     22 000030  000003                           bpt
     23                                 ;
     24 000032  066701  177750                   add     n_five,r1
     25 000036  000003                           bpt
     26                                 ;
     27 000040  066702  177742                   add     n_five,r2
     28 000044  066702  177736                   add     n_five,r2
     29 000050  000003                           bpt
     30                                 ;
     31 000052  066703  177726                   add     five,r3
     32 000056  066703  177722                   add     five,r3
     33 000062  000003                           bpt
     34                                 ;
     35 000064  066704  177714                   add     five,r4
     36 000070  066704  177712                   add     n_five,r4
     37 000074  066704  177706                   add     n_five,r4
     38 000100  000003                           bpt
     39                                 ;
     40 000102  066705  177700                   add     n_five,r5
     41 000106  066705  177672                   add     five,r5
     42 000112  066705  177666                   add     five,r5
     43 000116  000003                           bpt
     44                                 ;
     45 000120  000000                           halt
     46                                 ;
     47 000122  000010'                          .end    start


five      000004 R  
n_five    000006 R  
n_one     000002 R  
one       000000 R  
pc      = 000007 %  
r0      = 000000 %  
r1      = 000001 %  
r2      = 000002 %  
r3      = 000003 %  
r4      = 000004 %  
r5      = 000005 %  
r6      = 000006 %  
r7      = 000007 %  
sp      = 000006 %  
start     000010 RG 
