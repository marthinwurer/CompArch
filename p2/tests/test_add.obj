000 2 01 00
002 2 ff ff
004 2 05 00
006 2 fb ff
008 4 c1 6d f4 ff
00c 2 03 00
00e 4 c1 6d f2 ff
012 2 03 00
014 4 c1 6d ea ff
018 2 03 00
01a 4 c1 6d e8 ff
01e 2 03 00
020 4 c2 6d e2 ff
024 4 c2 6d de ff
028 2 03 00
02a 4 c3 6d d6 ff
02e 4 c3 6d d2 ff
032 2 03 00
034 4 c4 6d cc ff
038 4 c4 6d ca ff
03c 4 c4 6d c6 ff
040 2 03 00
042 4 c5 6d c0 ff
046 4 c5 6d ba ff
04a 4 c5 6d b6 ff
04e 2 03 00
050 2 00 00
008
