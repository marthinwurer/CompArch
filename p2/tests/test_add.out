CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 10
000010:  PS[00]  066701 177764  ADD 177764(R7),R1
000014:  PS[00]  000003  BPT
  R0[000000]  R1[000001]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000016]
000016:  PS[00]  066701 177762  ADD 177762(R7),R1
000022:  PS[00]  000003  BPT
  R0[000000]  R1[000006]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000024]
000024:  PS[00]  066701 177752  ADD 177752(R7),R1
000030:  PS[01]  000003  BPT
  R0[000000]  R1[000005]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000032]
000032:  PS[01]  066701 177750  ADD 177750(R7),R1
000036:  PS[05]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000040]
000040:  PS[05]  066702 177742  ADD 177742(R7),R2
000044:  PS[10]  066702 177736  ADD 177736(R7),R2
000050:  PS[11]  000003  BPT
  R0[000000]  R1[000000]  R2[177766]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000052]
000052:  PS[11]  066703 177726  ADD 177726(R7),R3
000056:  PS[00]  066703 177722  ADD 177722(R7),R3
000062:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[177766]  R3[000012]
  R4[000000]  R5[000000]  R6[000000]  R7[000064]
000064:  PS[00]  066704 177714  ADD 177714(R7),R4
000070:  PS[00]  066704 177712  ADD 177712(R7),R4
000074:  PS[05]  066704 177706  ADD 177706(R7),R4
000100:  PS[10]  000003  BPT
  R0[000000]  R1[000000]  R2[177766]  R3[000012]
  R4[177773]  R5[000000]  R6[000000]  R7[000102]
000102:  PS[10]  066705 177700  ADD 177700(R7),R5
000106:  PS[10]  066705 177672  ADD 177672(R7),R5
000112:  PS[05]  066705 177666  ADD 177666(R7),R5
000116:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[177766]  R3[000012]
  R4[177773]  R5[000005]  R6[000000]  R7[000120]
000120:  PS[00]  000000  HALT

Machine Halted - HALT instruction

Simulated time 191 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
